### PR TITLE
[noise support in MPS] MPO and Hamiltonian with leakage 

### PR DESCRIPTION
**Pull request by murogrande**
related to #179 
Previously, we work in MPS state class in order to support 3 level system. Now, it is the turn of MPO. 

- the tests regarding MPO and Rydberg Hamiltonian are updated

### CODE REVIEW

File: emu_mps/hamiltonian.py [hunk 1]
```diff
@@ -220,14 +232,14 @@ def left_factor(self, n: int) -> torch.Tensor:
         fac[0, :, :, 0] = Operators.id
```
[the remark this answers is not in the document]
Comment (murogrande) [reply emu_mps/hamiltonian.py]: this is the xy hamiltonian, let me update this also

File: test/emu_mps/test_hamiltonian.py [hunk 2]
```diff
@@ -306,34 +447,35 @@ def test_5_qubit(hamiltonian_type):
...
 
 
-@pytest.mark.parametrize(
-    "hamiltonian_type",
-    [
-        HamiltonianType.Rydberg,
-        HamiltonianType.XY,
-    ],
-)
-def test_9_qubit_noise(hamiltonian_type):
```
[the remark this answers is not in the document]
Comment (murogrande) [reply test/emu_mps/test_hamiltonian.py]: ok let me undo the changes

File: emu_mps/hamiltonian.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (murogrande) [reply emu_mps/hamiltonian.py]: >  - just keep the original implementation, but add an assert that dim == 2

I leave this since we are now testing for 3 level system the XY hamiltonian too

File: emu_base/jump_lindblad_operators.py [hunk 3]
```diff
@@ -2,43 +2,59 @@
...
     if noise_type == "dephasing":
         if noise_model.hyperfine_dephasing_rate != 0.0:
             raise NotImplementedError("hyperfine_dephasing_rate is unsupported")
 
         c = math.sqrt(noise_model.dephasing_rate / 2)
-        return [
-            torch.tensor([[-c, 0], [0, c]], dtype=torch.complex128),
-        ]
+        dephasing = torch.zeros(dim, dim, dtype=dtype)
+        print(dephasing)
```
[the remark this answers is not in the document]
Comment (murogrande) [reply emu_base/jump_lindblad_operators.py]: good catch!

File: emu_mps/hamiltonian.py [hunk 4]
```diff
@@ -416,20 +442,25 @@ def update_H(
...
+    assert noise.shape == (2, 2) or (3, 3)
     nqubits = omega.size(dim=0)
 
     a = torch.tensordot(omega * torch.cos(phi), Operators.sx, dims=0)
-    c = torch.tensordot(delta, Operators.pu, dims=0)
+    c = torch.tensordot(delta, Operators.n, dims=0)
     b = torch.tensordot(omega * torch.sin(phi), Operators.sy, dims=0)
 
-    single_qubit_terms = a + b - c + noise
+    # single_qubit_terms = a + b - c + noise
```
Comment (elmerhej) [inline emu_mps/hamiltonian.py]: delete this?

**Review by elmerhej — APPROVED** [review]
